common/src/IO/ObjSerializer.cpp

```diff
@@ -76,7 +77,9 @@ namespace TrenchBroom {
             }
 
             for (const std::string& texture : textureNames) {
-                std::fprintf(m_mtlStream, "newmtl %s\n", texture.c_str());
+				std::fprintf(m_mtlStream, "newmtl %s\n", texture.c_str());
+				std::fprintf(m_mtlStream, "map_Kd %s%s%s\n\n", m_gamePath.asString().c_str(), "\\textures\\", texture.c_str());
```

Comment:
You're assuming that textures are always stored relative to the game path, under `textures`. This might not be true in all cases:

- Mip textures are not stored individually on the filesystem at all, but are usually stored in wads.
- Wads can be located anywhere on the filesystem.
- We are going to drop the requirement that textures be stored under the game path in the future.

I think we need to reconsider how the `ObjFileSerializer` gets the absolute path of a texture in a reliable way, without making these assumptions. Textures know about their file name, but that doesn't give you the absolute path to the file they were loaded from - this requires accessing the texture collection that contains the texture. 

Currently, there is no way to determine which texture collection a texture belongs to. But `TextureManager` could determine this information. It doesn't do that now, but it could be extended so that it can.

I think the right way is to pass a mapping from texture (name) to path to the serializer in the constructor. This mapping could be a lambda that uses the texture manager to reconstruct the texture's path. Alternatively, we could store the path in the texture itself. @ericwa what do you think?